eng/scripts/Sdk-Generate.ps1

```diff
@@ -0,0 +1,173 @@
+        # replacing repo path
+        $line -replace "https:\/\/[^`"]+?$headSha", "$repoHttpsUrl/blob/$headSha"
+
+        $isUpdatedLines = $true
+        $sdkPaths += (get-item $path).Directory.Parent.FullName | Resolve-Path -Relative
+      }
+      else
+      {
+        $line
+      }
+    }
+    if($isUpdatedLines)
+    {
+      $updatedLines | Out-File -FilePath $path
+    }
+  }
+}
+
+Write-Host "Updated autorest.md files for all the changed swaggers. `n"
+
+function Test-PreviousScript() 
+{
+  $result = $null
+  if($LASTEXITCODE -eq 0)
+  {
+    $result = "succeeded"
+  }
+  else 
+  {
+    $result = "failed"
+  }
+  Write-Output $result
+}
+
+$sdkPaths = $sdkPaths | select -Unique
+$packages = @()
+$artifactsPath = "$RepoRoot/artifacts/packages"
+if(!(Test-Path $artifactsPath))
+{
+  New-Item -ItemType Directory -Force -Path $artifactsPath
+}
+foreach ($sdkPath in $sdkPaths)
+{
+  $packageName = Split-Path $sdkPath -Leaf
+  Write-Host "Generating code for " $packageName
+  $srcPath = Join-Path $sdkPath 'src'
+  dotnet msbuild /restore /t:GenerateCode $srcPath
+
+  $artifacts = @()
+  $hasBreakingChange = $null
+  $content = $null
+  $result = Test-PreviousScript
+  if($result -eq "succeeded")
+  {
+    Write-Host "Successfully generated code for" $packageName "`n"
+    
+    $csprojPath = Get-ChildItem $srcPath -Filter *.csproj -Recurse
+    dotnet pack $csprojPath --output $artifactsPath /p:RunApiCompat=$false
+    $result = Test-PreviousScript
+    if($result -eq "succeeded")
+    {
+      $artifacts +=  Get-ChildItem $artifactsPath -Filter *$packageName* -Recurse | Select-Object -ExpandProperty FullName | Resolve-Path -Relative
+
+      $logFilePath = Join-Path "$srcPath" 'log.txt'
+      if(!(Test-Path $logFilePath))
+      {
+        New-Item $logFilePath
+      }
+      dotnet build $csprojPath /t:RunApiCompat /p:TargetFramework=netstandard2.0 /flp:v=m`;LogFile=$logFilePath
+      
+      $result = Test-PreviousScript
+      if($result -eq "succeeded")
+      {
+        $hasBreakingChange = $false
+      }
+      else
+      {
+        $logFile = Get-Content -Path $logFilePath | select-object -skip 2
+        $breakingChanges = foreach($Obj in $logFile) 
+        {       
+          $begin = ""
+          $end = ",`n"
+          $begin + $Obj + $end
+        }
+        $content = "Breaking Changes: $breakingChanges"
+        $hasBreakingChange = $true
+        $result = "succeeded"
+      }
+      if (Test-Path $logFilePath) 
+      {
+        Remove-Item $logFilePath
+      }
+    }
+  } 
+  else 
+  {
+    Write-Host "Error occurred while generating code for" $packageName "`n"
+  }
+
+  $path = @()
+  $path += $sdkPath
+
+  $packageNameArr = $packageName.Split(".")
+  $name = $input.relatedReadmeMdFiles -match $packageNameArr[2]
+  $readmeMd = @()
+  if($packageName -match "Azure.ResourceManager")
+  {
+    $readmeMd += $name -match "resource-plane"
```

Comment:
These path names might not be something we should rely on. Don't we know which readme's based on the matching input files in the readmes?

Reply:
I've updated the [script](https://github.com/openapi-env-test/azure-sdk-for-net/blob/05c4071e24f35114137b5527864454ba11023235/eng/scripts/Sdk-Generate.ps1#L37) and added readme's based on the matching readme input files but now in order to add readme path in output json, it expects readme.md as an input file in autorest.md. Currently for few SDKs we use directly spec(json file) as an input file. I have already created an [issue ](https://github.com/Azure/azure-sdk-for-net/issues/16445)to use readmes in autorest.md.